aptos-move/framework/aptos-framework/sources/TestCoin.move

```diff
@@ -47,44 +30,34 @@ module AptosFramework::TestCoin {
         inner: vector<DelegatedMintCapability>,
     }
 
-    /// Events handles.
-    struct TransferEvents has key {
-        sent_events: EventHandle<SentEvent>,
-        received_events: EventHandle<ReceivedEvent>,
-    }
+    public fun initialize(core_framework: &signer, core_resource: &signer): BurnCapability<TestCoin> {
+        SystemAddresses::assert_core_resource(core_resource);
 
-    struct SentEvent has drop, store {
-        amount: u64,
-        to: address,
-    }
+        let (mint_cap, burn_cap) = Coin::initialize_with_capabilities<TestCoin>(
+            core_framework,
+            ASCII::string(b"Test Coin"),
+            ASCII::string(b"TC"),
+            6, /* decimals */
+            false, /* monitor_supply */
+        );
 
-    struct ReceivedEvent has drop, store {
-        amount: u64,
-        from: address,
-    }
+        // Mint the core resource account TestCoin for gas so it can execute system transactions.
+        Coin::register_for_account<TestCoin>(core_resource);
+        let coins = Coin::mint_with_capability<TestCoin>(
+            18446744073709551615,
```

Comment:
what happens if we run out of all testcoin?

Reply:
In testnet, we do have the key for the core resources account, which has mint capability so we can mint more to it. With that said, there might be a way to avoid having to make transactions from that account, but I'll need to do separate research on that.

Reply:
this account is only paying gas to create new accounts

Reply:
I'm not sure if there's anything actionable for now. We can discuss/look into potentially removing this or at least minting less separately.